src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -29,16 +31,39 @@ public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorksp
             // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
             var extensionManager = workspace.GetVsService<SVsExtensionManager, IVsExtensionManager>();
 
-            var builder = ImmutableArray.CreateBuilder<HostDiagnosticAnalyzerPackage>();
-            foreach (var extension in extensionManager.GetEnabledExtensions(AnalyzerContentTypeName))
+            // get rootfolder and shellfolder location
+            string rootFolder;
+            string shellFolder;
+            if (TryGetRootAndShellFolder(extensionManager, out shellFolder, out rootFolder))
```

Comment:
Yes, its just making an existing API public. If the shell team is busy, even one of us can make that change and get it reviewed from them? Lets see what their triage team decides on the bug..